nav2_amcl/src/amcl_node.cpp

```diff
@@ -1053,31 +1051,12 @@ AmclNode::handleMapMessage(const nav_msgs::msg::OccupancyGrid & msg)
       global_frame_id_.c_str());
   }
 
-  freeMapDependentMemory();
-  // Clear queued laser objects because they hold pointers to the existing
-  // map, #5202.
-  lasers_.clear();
-  lasers_update_.clear();
-  frame_to_laser_.clear();
-
   map_ = convertMap(msg);
 
 #if NEW_UNIFORM_SAMPLING
   createFreeSpaceVector();
 #endif
-  // Create the particle filter
-  initParticleFilter();
-  // resample_count = 0 is extra
-
-  // Instantiate the sensor objects
-  motion_model_.reset();
-  motion_model_ = std::unique_ptr<nav2_amcl::MotionModel>(nav2_amcl::MotionModel::createMotionModel(
-        robot_model_type_, alpha1_, alpha2_, alpha3_, alpha4_, alpha5_));
 
-  // Laser
-  lasers_.clear();
-
-  handleInitialPose(last_published_pose_);
```

Comment:
isn't this necessary?

Reply:
When `initialPoseReceived` is called, it will call `handleInitialPose`. I don't see any reason on why `handleMapMessage` should handle the initial pose.

Reply:
It seems like in navigation1 there was some pose handling on new mapCb https://github.com/ros-planning/navigation/blob/melodic-devel/amcl/src/amcl_node.cpp#L1574

I'm not sure if related but I'm not seeing an analog here

Reply:
So far we have been setting the initial pose from rviz, but we need to also be able to handle it if it comes programmatically before the first map is ever loaded.  I think we are missing this, we should open an issue to support.